core/includes/install.core.inc

```diff
@@ -893,6 +898,8 @@ function install_verify_database_utf8mb4() {
 
 /**
  * Verifies the existing settings in settings.php.
+ *
+ * @return TRUE|array of errors|FALSE
  */
 function install_verify_settings() {
```

Comment:
Instead of modifying the return value, we could add a pass-by-value array of $errors here. `function install_verify_settings(&$errors) {`. Then the return signature would stay the same.

Reply:
Don't have strong feelings about changing the return signature (`install_verify_settings` is only used here) or pass a reference to a `$errors` array. 
I wonder what is the most common pattern. According to `git grep`, this is `return $errors`:
```bash
$ git grep '&$errors' |wc -l
12
$ git grep 'return $errors' |wc -l
32
```

Reply:
@opi Even though `return $errors` seems to be more common, it sounds like Nate thinks the pass-by-value is better option.